### PR TITLE
Define and use a setting that controls how long authentication documents are cached.

### PR DESCRIPTION
**Pull request by leonardr**
## Description

This branch gives us the power to fine-tune or disable the caching of authentication documents. This was introduced for performance reasons but it makes certain deployment problems difficult to test, so it should be possible to disable it.

## Motivation and Context

https://jira.nypl.org/browse/SIMPLY-3548

## How Has This Been Tested?

In addition to the automated tests, I tested this with a local database by setting the timeout to 10 seconds, or 0 seconds, and adding a debug statement to the code to signal whenever a fresh auth document was created.

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/controller.py [hunk 1]
```diff
@@ -538,9 +543,15 @@ def authentication_for_opds_document(self):
         OPDS document in the cache, then return the cached version.
         """
         name = flask.request.library.short_name
-        if name not in self.authentication_for_opds_documents:
-            self.authentication_for_opds_documents[name] = self.auth.create_authentication_document()
-        return self.authentication_for_opds_documents[name]
+        value = self.authentication_for_opds_documents.get(name, None)
+        if value is None:
```
Comment (EdwinGuzman) [inline api/controller.py:547]: This is similar to how it was before except that now it's checking if it's `None`. Why the change to this syntax?

**Review by EdwinGuzman — COMMENTED** [review]
This looks good but the PR branch Travis test is failing.

File: api/controller.py (hunk 1 above)
Comment (leonardr) [reply api/controller.py:547]: The old code won't work if the caching time is set to zero. I create the authentication document and stick it into `self.authentication_for_opds_documents`, where it is _immediately deleted_. So `self.authentication_for_opds_documents[name]` fails even though I just put the document into there.

Instead I make sure the document (whether retrieved from cache or created fresh) is stored in `value`. I stick `value` in the cache for next time, but it doesn't matter if it immediately gets deleted; I already have it.

**Comment by leonardr** [comment]
Yeah, I don't know what to do about that. I can't use Travis to restart the build; it fails with an unspecified error.

**Comment by leonardr** [comment]
Made a cosmetic change to force the tests to run again and this time it worked.
